src/task.rs

```diff
@@ -110,17 +127,15 @@ impl Task {
         }
     }
 
-    //TODO: The current way this is done may not be entirely accurate for tasks that can be done on
-    //multiple days within certain time bounds.
     fn calculate_flexibility(&mut self) {
-        let mut hours_available = 0;
+        let mut flexibility = 0;
         for slot in &self.slots {
-            hours_available += slot.num_hours();
+            flexibility += slot.num_hours() - self.duration + 1;
```

Comment:
```suggestion
            flexibility += std::cmp::max(0, slot.num_hours() - self.duration + 1);
```

Reply:
`calculate_flexibility` is called after `remove_invalid_slots` has been called so `slot.num_hours() - self.duration + 1` should never be less than 0. If it is, wouldn't we rather be alerted to this by a `panic`?

Reply:
Yup, agree!